src/data/blob_tests.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 MaidSafe.net limited.
```

Comment:
This file isn't really that big, and I think it could just be a sub-module in `data/blob.rs`: `mod tests { ... }`. Not really an issue, but it would make the directory structure a bit simpler.

Reply:
It's for consistency, because `map.rs` and `sequence.rs` will have `map_tests.rs` and `sequence_tests.rs` in `data` module. Just not part of this PR.

Also I think there should be more tests there, looks a bit thin?

Reply:
Agreed that the types can have more tests.
Having a `tests` module in each map, sequence and blob seems to be a good idea.

Another option would be to have a `tests`module with separate files for each of the types. This is usually done when there are a _lot_ of tests. If that's the case then we can have `data/tests/blob.rs` and similar files for map and sequence. But if the tests are minimal then a sub-module should be fine.

Reply:
Gotcha @oetyng and agreed with Lionel: I like `tests/blob.rs` more than `blob_tests.rs`. The former is what you'd more typically see in a Rust project.